asset-registry/src/lib.rs

```diff
@@ -0,0 +1,235 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::EnsureOrigin;
+use frame_support::weights::constants::WEIGHT_PER_SECOND;
+use frame_support::{log, pallet_prelude::*};
+use frame_system::pallet_prelude::*;
+pub use module::*;
+use orml_traits::asset_registry::{FixedConversionRateProvider, WeightToFeeConverter};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{AtLeast32BitUnsigned, Member},
+	DispatchResult,
+};
+use sp_std::prelude::*;
+use xcm::latest::prelude::*;
+use xcm_builder::TakeRevenue;
+use xcm_executor::{traits::WeightTrader, Assets};
+
+pub trait AssetIdAssigner<AssetId, AssetMetadata> {
+	fn assign_id(asset_metadata: &AssetMetadata, last_id: Option<AssetId>) -> AssetId;
```

Comment:
make it something like `process_asset(id: Option<AssetId>, metadata: AssetMetadata) -> Result<(AssetId, Metadata), DispatchError>` will be more flexible

not really need to pass last id. the implementation can just read it from this pallet

Reply:
Done. However, without the last id, we can't provide a default `SequentialId` implementation - each chain will need to write an explicit implementation now. Because of that, I would actually be in favor of adding it back in as an arg. What do you think?

Reply:
The default SequentialId can still read the last id from the pallet?